gdk: trust nullability annotations

bilelmoussaoui opened the pull request:
Those were the ones that i noticed and needs fixes upstream
- gdk_content_deserializer_get_cancellable: needs a nullable annotation
- gdk_content_serializer_get_cancellable: needs a nullable annotation
- gdk_event_get_surface: needs a nullable annotation
- gdk_popup_get_parent: needs a nullable annotation

sdroege commented on gdk4/Gir.toml:
```diff
@@ -243,27 +243,19 @@ generate_builder = false
     [[object.function]]
+    name = "get_cancellable"
+        [object.function.return]
+        nullable = true # TODO: fixme upstream
```
Please create an issue before merging

sdroege replied:
Also for the others

bilelmoussaoui replied:
Sent a PR https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4310

sdroege replied:
Can you put a link to that into `Gir.toml`?

bilelmoussaoui replied:
I don't plan on merging it yet, still working on the gtk ones

sdroege commented:
Looks generally correct otherwise

bilelmoussaoui commented:
Okay, I have dropped all the `trust_return_value_nullability = false`, regenerated type by type; checked the corresponding C function and updated https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4310 with whatever fixes that are needed.

Once that merge request is merged, we can merge this one & drop the manual annotations here once we get a new gir file.

bilelmoussaoui commented:
The merge request got merged, I went ahead and done something similar for the constructors as well https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4315. This should be ready to be merged.
